### PR TITLE
Feature/gal 804- User Authentication: Error State

### PR DESCRIPTION
**Pull request by Amar-NHS**
## Description

Added Authorisation checks according to the Acceptance Criteria on the ticket and added redirecting to the relevant error pages. 

1.  Given the account does not exist or is inactive - user will be redirected to the log out error page with error 'There is no record of this account'
2. Given the Galleri activity code missing or authentication is not L3(This will be set to L3 after moving to CIS2 Integration environment) - user will be redirected to the log out error page with error 'You do not have the required access for this system'

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -19,8 +21,11 @@ try {
   console.error("Error parsing USERS environment variable:", error);
 }
 
+// Environment Variables
 const GET_USER_ROLE = process.env.NEXT_PUBLIC_GET_USER_ROLE;
 const ENVIRONMENT = process.env.NEXT_PUBLIC_ENVIRONMENT;
+const GALLERI_ACTIVITY_CODE = process.env.GALLERI_ACTIVITY_CODE;
```
Comment (avi312NHS) [inline src/app/api/auth/[...nextauth]/route.ts:27]: is this going in as an array? if it is then 
`process.env.GALLERI_ACTIVITY_CODES` and `process.env.GALLERI_ACTIVITY_NAME`

File: src/app/page.jsx [hunk 2]
```diff
@@ -15,14 +14,14 @@ export default function Root() {
   return (
     <div>
       <body className="js-enabled">
-        <a
+        {/* <a
```
Comment (avi312NHS) [inline src/app/page.jsx:17]: can we delete this?

File: src/app/api/auth/[...nextauth]/route.ts (hunk 1 above)
Comment (Amar-NHS) [reply src/app/api/auth/[...nextauth]/route.ts:27]: No those two environment variables will be going in as strings.

**Comment by avi312NHS** [comment]
Looks good 👍

**Review by avi312NHS — APPROVED** [review]

**Review by Abdul-NHS — APPROVED** [review]

**Pull request merged**